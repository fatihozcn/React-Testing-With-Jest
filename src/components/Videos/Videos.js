import React from 'react'
const Videos = ({ videos = [] }) => {
        if(videos.length<=0) {
        return <h2>Sorry, no videos found.</h2>;
    }
    return (
    <div>
        {videos.map((video, index)=>(
            <h4 key={index}>{video}</h4>
        ))}
    </div>
    )
}

export default Videos;
