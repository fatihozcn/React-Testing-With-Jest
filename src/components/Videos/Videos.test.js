import React from "react";
import renderer from "react-test-renderer";
import Videos from './Videos'

test('Videos Empty Renders Sorry, no videos found', ()=>{
    const tree = renderer.create(<Videos />).toJSON();
    expect(tree).toMatchSnapshot();
})

test('Videos 1 Item Renders Correctly', ()=>{
    const tree = renderer.create(<Videos videos={["Exxen"]}/>).toJSON();
    expect(tree).toMatchSnapshot();
})

test('Videos 3 Items Renders Correctly', ()=>{
    const tree = renderer.create(<Videos videos={["Udemy", "Youtube", "Tutorial"]}/>).toJSON();
    expect(tree).toMatchSnapshot();
})