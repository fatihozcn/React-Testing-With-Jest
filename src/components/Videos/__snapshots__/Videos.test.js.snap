// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Videos 1 Item Renders Correctly 1`] = `
<div>
  <h4>
    Exxen
  </h4>
</div>
`;

exports[`Videos 3 Items Renders Correctly 1`] = `
<div>
  <h4>
    Udemy
  </h4>
  <h4>
    Youtube
  </h4>
  <h4>
    Tutorial
  </h4>
</div>
`;

exports[`Videos Empty Renders Sorry, no videos found 1`] = `
<h2>
  Sorry, no videos found.
</h2>
`;
